Make format-fix throws a FileNotFoundError: No such file or directory: 'black'

### What happens?

When running make format-fix on master an `FileNotFoundError` is thrown:

```
File "/opt/homebrew/Cellar/python@3.11/3.11.4_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 1950, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'black'
make: *** [format-fix] Error 1
```

### To Reproduce

run `make format-fix` without package `black` installed

### OS:

MacOS

### DuckDB Version:

master

### DuckDB Client:

cli

### Full Name:

Lotte Felius

### Affiliation:

CWI

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -70,8 +70,8 @@ This project and everyone participating in it is governed by a [Code of Conduct]
 
 * Use tabs for indentation, spaces for alignment.
 * Lines should not exceed 120 columns.
-* To make sure the formatting is consistent, please use version 11.0.1, installable through `python3 -m pip install clang-format==11.0.1`
-* `clang_format` enforces these rules automatically, use `make format-fix` to run the formatter.
+* To make sure the formatting is consistent, please use version 10.0.1, installable through `python3 -m pip install clang-format==10.0.1.1`
+* `clang_format` and `black` enforce these rules automatically, use `make format-fix` to run the formatter.
 * The project also comes with an [`.editorconfig` file](https://editorconfig.org/) that corresponds to these rules.
 
 ## C++ Guidelines

--- a/scripts/format.py
+++ b/scripts/format.py
@@ -15,15 +15,13 @@ from importlib.metadata import version
 
 try:
     import_module('black')
-except ImportError:
-    print('you need to run `pip install black`')
+except ImportError as e:
+    print('you need to run `pip install black`', e)
     exit(-1)
 
-try:
-    import_module('clang_format')
-    assert version('clang_format').startswith('10.')
-except (ImportError, AssertionError):
-    print('you need to run `pip install clang_format==10.0.1.1`')
+ver = subprocess.check_output(('clang-format', '--version'), text=True)
+if '10.' not in ver:
+    print('you need to run `pip install clang_format==10.0.1.1 - `', ver)
     exit(-1)
 
 cpp_format_command = 'clang-format --sort-includes=0 -style=file'

--- a/scripts/format.py
+++ b/scripts/format.py
@@ -10,6 +10,21 @@ import subprocess
 import difflib
 import re
 from python_helpers import open_utf8
+from importlib import import_module
+from importlib.metadata import version
+
+try:
+    import_module('black')
+except ImportError:
+    print('you need to run `pip install black`')
+    exit(-1)
+
+try:
+    import_module('clang_format')
+    assert version('clang_format').startswith('10.')
+except (ImportError, AssertionError):
+    print('you need to run `pip install clang_format==10.0.1.1`')
+    exit(-1)
 
 cpp_format_command = 'clang-format --sort-includes=0 -style=file'
 cmake_format_command = 'cmake-format'

